webhooks/clusterclass.go

```diff
@@ -99,6 +103,19 @@ func (webhook *ClusterClass) ValidateUpdate(_ context.Context, oldObj, newObj ru
 
 // ValidateDelete implements validation for ClusterClass delete.
 func (webhook *ClusterClass) ValidateDelete(ctx context.Context, obj runtime.Object) error {
+	clusterClass, ok := obj.(*clusterv1.ClusterClass)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a ClusterClass but got a %T", obj))
+	}
+	clusters, err := webhook.clustersUsingClusterClass(clusterClass)
+	if err != nil {
+		return err.ToAggregate()
+	}
+	if len(clusters) != 0 {
+		// TODO: Improve error here to include the names of some clusters using the clusterClass.
```

Comment:
Should this be done in this PR? (e.g. just pick (at most) the first 5 and add them with `.e.g` to the error message